### ISSUE TITLE
Queue length is wrong when using Redis and task priorities

### ISSUE THREAD

**Issue by ArchFrosty** [danihodovic/celery-exporter#272]
tldr; only the priority 0 queue is counted

This one is fairly simple:
when you get active queues here
https://github.com/danihodovic/celery-exporter/blob/c837bab76a2cf40facc31a2b0ae687720bf3211a/src/exporter.py#L229
Celery will indeed return you the active queues from its point of view - for example the default queue `'celery'`. The problem is that when using Redis, priority is implemented by actually crating multiple queues in Redis.
By default it creates 3 additional queues:  `'celery\x06\x163'`  `'celery\x06\x166'`  `'celery\x06\x169'` - queues 3 6 and 9 with `'\x06\x16'` as separator between name and priority number. Priority numbers used when submitting tasks are then mapped to these queues (0 highest prio and 9 lowest), the highest priority queue (with priority value 0) **is just called its name, in this case `'celery'` without separator and number**

So then when you eventually call 
https://github.com/danihodovic/celery-exporter/blob/c837bab76a2cf40facc31a2b0ae687720bf3211a/src/exporter.py#L250
and eventually call llen
https://github.com/danihodovic/celery-exporter/blob/c837bab76a2cf40facc31a2b0ae687720bf3211a/src/exporter.py#L449-L450
 on the name of this queue, you only get the tasks in the priority 0 queue.

Also the separator and then number of queues is configurable.
https://docs.celeryq.dev/en/stable/userguide/routing.html#redis-message-priorities

**Comment by danihodovic** [comment]
https://github.com/danihodovic/celery-exporter/pull/204#issuecomment-1545974112

tl;dr I don't think I will add queue length for Redis.

**Comment by zdenekbauer** [comment]
Hi, we have the same problem. I saw that there was already prepared pull-request for that, can you please consider it again? It would be amazing to have this feature.

**Comment by paulkiernan** [comment]
Bummer. We use redis priority queues and would benefit from this.

**Comment by ArchFrosty** [comment]
We ended up moving to RabbitMQ - this was one of the main reasons, but there were other ones too such as better replication and failover. In retrospective it was 100% the right decision.
